File: java/exercises/src/test/java/io/github/ajoz/workshop/fp/java/tests/theory/Between.java
package io.github.ajoz.workshop.fp.java.tests.theory;

import org.junit.experimental.theories.ParametersSuppliedBy;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@ParametersSuppliedBy(BetweenSupplier.class)
public @interface Between {
    int first();
    int last();
}

File: kotlin/exercises/src/test/kotlin/io/github/ajoz/workshop/fp/kotlin/tests/theory/Between.kt
package io.github.ajoz.workshop.fp.kotlin.tests.theory

import org.junit.experimental.theories.ParametersSuppliedBy

@Retention(AnnotationRetention.RUNTIME)
@ParametersSuppliedBy(BetweenSupplier::class)
annotation class Between(
        val first: Int,
        val last: Int
)
